sw/device/silicon_creator/rom_ext/rom_ext.c

```diff
@@ -197,8 +202,48 @@ static rom_error_t boot_svc_next_boot_bl0_slot_handler(
   return kErrorRomExtBootFailed;
 }
 
+OT_WARN_UNUSED_RESULT
+static rom_error_t boot_svc_primary_boot_bl0_slot_handler(
+    boot_svc_msg_t *boot_svc_msg, boot_data_t *boot_data) {
+  uint32_t active_slot = boot_data->primary_bl0_slot;
+  uint32_t requested_slot = boot_svc_msg->primary_bl0_slot_req.primary_bl0_slot;
+
+  // In cases where the primary is already set to the requested slot, this
+  // function is a no-op.
+  if (active_slot != requested_slot) {
+    switch (launder32(requested_slot)) {
+      case kBootDataSlotA:
+        HARDENED_CHECK_EQ(requested_slot, kBootDataSlotA);
+        boot_data->primary_bl0_slot = kBootDataSlotA;
+        break;
+      case kBootDataSlotB:
+        HARDENED_CHECK_EQ(requested_slot, kBootDataSlotB);
+        boot_data->primary_bl0_slot = kBootDataSlotB;
+        break;
+      default:
+        HARDENED_TRAP();
+        OT_UNREACHABLE();
+    }
+
+    // Write boot data, updating relevant fields and recomputing the digest.
+    HARDENED_RETURN_IF_ERROR(boot_data_write(boot_data));
+    // Read the boot data back to ensure the correct slot is booted this time.
+    HARDENED_RETURN_IF_ERROR(boot_data_read(lc_state, boot_data));
+    HARDENED_RETURN_IF_ERROR(boot_data_check(boot_data));
```

Comment:
Reading back and using the actual data stored in the flash is a great idea!

But let's remove the digest check here since we have a separate task to harden `boot_data_write()`.
```suggestion
```